server/src/com/thoughtworks/go/server/service/UserService.java

```diff
@@ -258,6 +259,24 @@ public void deleteUser(String username, HttpLocalizedOperationResult result) {
         }
     }
 
+    public HashMap<String, List<String>> deleteUsers(List<String> userNames, HttpLocalizedOperationResult result) {
+        if( userNames == null || userNames.isEmpty()){
+            result.notFound(LocalizedMessage.string("NO_USERS_SELECTED"), HealthStateType.notFound());
+            return null;
+        }
+        try {
+            userDao.deleteUsers(userNames);
+            result.setMessage(LocalizedMessage.string("RESOURCE_DELETE_SUCCESSFUL", "users", userNames));
+        } catch (BulkDeleteException e) {
+            result.unprocessableEntity(LocalizedMessage.string("USER_ENABLED_OR_NOT_FOUND", e.getEnabledUsers(), e.getNullUsers()));
```

Comment:
Instead of relying on an exception, perhaps you want to return a `DeletionResult` object that can answer whether the delete `isSuccessful()` and if there are errors, then you can ask for `enabledUsers` and `nonExistantUsers` as required.